### DISCUSSION TITLE
Problem with RightToLeft text

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#15071] [Q&A]
I load in a mixed English/Urdu text. The Urdu characters render individually but appear left to right. It would appear that the RTF `\rtlch` flag isn't being properly understood.
In our older, VB6-based, editor I was able to translate 
![image](https://github.com/unoplatform/uno/assets/209621/2b2bf2a7-9892-478c-9a10-ef350ddaecea)
into
![image](https://github.com/unoplatform/uno/assets/209621/e652e149-ad58-45d5-a13f-14eea5b4e3c2)

Currently I'm having considerable difficulty convincing Uno4 to behave in a similar manner. I set `FlowDirection` to `RightToLeft` but that doesn't appear to affect anything. Similarly, setting `TextAlignment` to `DetectFromContent` didn't appear to do anything, and if `\rtlch` isn't being properly recognized ...

The closest I've gotten to the preferred is
![image](https://github.com/unoplatform/uno/assets/209621/aa84702b-66c5-4c73-9f62-87e685ae0acc)

Am not sure what to go from here. Will upgrading to Uno5 help? Would creating a custom RichText object based on another 3rd party one that offer more functionality be possible? 

_LATER_

Maybe specifying the font would help a little.

**Comment by jeromelaban** [comment]
Thanks for the report. To be sure, on which platform head are you running for the screenshots? (make sure to specify it for every post you make)

**Comment by axtens** [comment]
Windows 11. But it seems that the issue is closer to home. I was assuming that the font rendering was at fault but have since found out that I was supplied with incorrect data.
